src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,123 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { getTodos } from '../../api';
+import { actions as todosActions } from '../../features/todos';
+import { useAppSelector } from '../../app/hooks';
+import { Todo } from '../../types/Todo';
+import { actions as selectedTodoActions } from '../../features/currentTodo';
+import { Loader } from '../Loader';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useDispatch();
+  const selectedTodo = useAppSelector((state) => state.currentTodo);
+  const filterStatus = useAppSelector((state) => state.filter.status);
+  const todos = useAppSelector((state) => state.todos);
+  const [isLoading, setIsLoading] = useState(false);
+  const query = useAppSelector((state) => state.filter.query);
+  const [isError, setIsError] = useState(false);
+  const setSelectedTodo = (todo: Todo) => {
+    dispatch(selectedTodoActions.setTodo(todo));
+  };
+
+  const filteredTodos = useMemo(() => {
+    const newTodos = query
+      ? todos.filter((todo) => todo.title.toLowerCase().includes(query))
+      : [...todos];
+
+    switch (filterStatus) {
+      case 'active':
+        return newTodos.filter((todo) => todo.completed === false);
+
+      case 'completed':
+        return newTodos.filter((todo) => todo.completed === true);
```

Comment:
```suggestion
        return newTodos.filter((todo) => !todo.completed);

      case 'completed':
        return newTodos.filter((todo) => todo.completed);
```
do not compare two boolean values. it is redundant)